### ISSUE TITLE
Docker daemon crashes under heavy file system access through mounted file system

### ISSUE THREAD

**Issue by bylexus** [docker/for-mac#1057]
May be related to: #1040 

### Expected behavior
docker container should manage also heavy file system access on local mounted files.

### Actual behavior
After a while (?) the docker daemon including all running container crashes, if heavy file system access occurs on a locally mounted file system (e.g. web server container with mounted user dir).

#### Diagnostig output:

Docker for Mac: version: 1.13.0-rc3-beta33.1 (e35ff41ca)
OS X: version 10.12.1 (build: 16B2555)
logs: /tmp/7A061A8D-C46F-48AF-A000-2C63F07B330F/20161219-143841.tar.gz
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     menubar
[OK]     disk

#### Output from syslog -w -k Sender Docker

$ syslog -w -k Sender Docker
-- before the crash
Dec 19 14:28:09 hostname Docker[32649] <Error>: Socket.Stream: caught socket is not connected
Dec 19 14:28:09 hostname Docker[32648] <Error>: transfused: read /Users: short read length 2
Dec 19 14:28:09 hostname Docker[32648] <Error>: Fatal unexpected exception: Socket.Closed
Dec 19 14:28:09 hostname Docker[32649] <Error>: Socket.Stream: caught socket is not connected
--- last message repeated 1 time ---
Dec 19 14:28:10 hostname Docker[23054] <Notice>: Reap com.docker.osxfs (pid 32648): exit status 1
Dec 19 14:28:11 hostname Docker[23054] <Notice>: Stop 1 children with order 1: com.docker.driver.amd64-linux (pid 32650)
Dec 19 14:28:11 hostname Docker[23054] <Notice>: Signal terminated to com.docker.driver.amd64-linux (pid 32650)
Dec 19 14:28:11 hostname Docker[23054] <Notice>: len(these)=1
Dec 19 14:28:11 hostname Docker[32650] <Notice>: sending SIGTERM to com.docker.hyperkit pid 32655
Dec 19 14:28:11 hostname Docker[23064] <Notice>: VM shutdown at 2016-12-19 14:28:11 +0100 CET
Dec 19 14:28:11 hostname Docker[23054] <Notice>: len(these)=1
--- last message repeated 4 times ---
Dec 19 14:28:17 hostname Docker[32650] <Notice>: virtio-net-vpnkit: initialising, opts="uuid=50b22a5d-3d7f-47f4-858e-dab24ce8ceda,path=/Users/username/Library/Containers/com.docker.docker/Data/s50,macfile=/Users/username/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/mac.0"
Dec 19 14:28:17 hostname Docker[32650] <Notice>: Interface will have uuid 50b22a5d-3d7f-47f4-858e-dab24ce8ceda
Dec 19 14:28:17 hostname Docker[32650] <Notice>: Connection established with MAC=c0:ff:ee:c0:ff:ee and MTU 1500
Dec 19 14:28:17 hostname Docker[32650] <Notice>: virtio-9p: initialising path=/Users/username/Library/Containers/com.docker.docker/Data/s40,tag=db
Dec 19 14:28:17 hostname Docker[32650] <Notice>: virtio-9p: initialising path=/Users/username/Library/Containers/com.docker.docker/Data/s51,tag=port
Dec 19 14:28:17 hostname Docker[32650] <Notice>: linkname /Users/username/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/tty
Dec 19 14:28:17 hostname Docker[32650] <Notice>: COM1 connected to /dev/ttys000
Dec 19 14:28:17 hostname Docker[32650] <Notice>: COM1 linked to /Users/username/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/tty
Dec 19 14:28:17 hostname Docker[23054] <Notice>: len(these)=1
Dec 19 14:28:17 hostname Docker[32650] <Notice>: Hypervisor subprocess has shutdown
Dec 19 14:28:17 hostname Docker[32650] <Notice>: TRIM is enabled but auto-compaction disabled: compacting now
Dec 19 14:28:17 hostname Docker[23064] <Notice>: VM shutdown at 2016-12-19 14:28:17 +0100 CET
Dec 19 14:28:17 hostname Docker[23054] <Notice>: len(these)=1
--- last message repeated 2 times ---
Dec 19 14:28:17 hostname Docker[23054] <Notice>: Reap com.docker.driver.amd64-linux (pid 32650): exit status 0
Dec 19 14:28:17 hostname Docker[23054] <Notice>: len(these)=0
Dec 19 14:28:17 hostname Docker[23054] <Notice>: Stop 2 children with order 2: com.docker.osxfs (pid 32648), com.docker.slirp (pid 32649)
Dec 19 14:28:17 hostname Docker[23054] <Notice>: Signal terminated to com.docker.slirp (pid 32649)
Dec 19 14:28:17 hostname Docker[23054] <Notice>: len(these)=1
Dec 19 14:28:17 hostname Docker[23054] <Notice>: Reap com.docker.slirp (pid 32649): signal: terminated
Dec 19 14:28:18 hostname Docker[23054] <Notice>: len(these)=0
-- end crash

### Steps to reproduce the behavior

Scenario:
- docker container with Web server and linked file system: /Users/<username>/www:/var/www
- web site requesting a lot of files in parallel (e.g. through a web site that loads a lot of resources)

1. Open a website that requests a lot of local files
2. reload some times
3. docker daemon crashes.

**Comment by djs55** [comment]
I also suspect this is related to #1040.

There are some relevant bugfixes scheduled for beta 34 due out tomorrow.

**Comment by dsheets** [comment]
This was fixed in Beta 34. Please open a new issue if you continue to experience problems.
